packages/hooks/use-aria-button/src/index.ts

```
@@ -104,7 +107,14 @@ export function useAriaButton(
 
   let isMobile = isIOS() || isAndroid();
 
-  if (deprecatedOnClick && typeof deprecatedOnClick === "function") {
+  if (
+    deprecatedOnClick &&
+    typeof deprecatedOnClick === "function" &&
+    // bypass since onClick is passed from <Link as={Button} /> internally
+    role !== "link" &&
+    // bypass since onClick is passed from useDisclosure's `getButtonProps` internally
+    !(props.hasOwnProperty("aria-controls") && props.hasOwnProperty("aria-controls"))
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)